controller/src/main/java/org/jboss/as/controller/operations/global/ReadResourceHandler.java

```diff
@@ -516,7 +523,13 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
                     PathElement path = entry.getKey();
                     ModelNode value = entry.getValue();
                     iter.remove();
+                    ModelNode responseHeaders = entry.getValue().get(RESPONSE_HEADERS).get(WARNINGS);
                     if (!value.has(FAILURE_DESCRIPTION)) {
+                        if (responseHeaders.isDefined()) {
+                            for (ModelNode response : responseHeaders.asList()) {
+                                context.addResponseWarning(Level.WARNING, response.get(WARNING));
```

Comment:
I guess that ideally instead of adding the Level.WARNING, we should take the level added to the warning ModelNode instead: `response.get(LEVEL)`

However, that will force you to parse later the String into a valid Level object, and looking at the code at https://github.com/wildfly/wildfly-core/blob/main/controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java#L384C5-L386 we only have such a level for a Warning. I'm just adding this in case any one could be asking for the same.

Reply:
I correct myself, indeed the warning could be logged with a lower level, so we will need to parse it from the `response.get(LEVEL)` value instead of hardcoding it always with `Level.WARNING`

Reply:
Done